tests/parametric/conftest.py

```diff
@@ -413,13 +411,77 @@ def get_open_port():
     return port
 
 
-@pytest.fixture
-def apm_test_server(request, library_env, test_id):
-    # Have to do this funky request.param stuff as this is the recommended way to do parametrized fixtures
-    # in pytest.
-    apm_test_library = _libs[context.scenario.library.library]
+@pytest.fixture(scope="session")
+def session_log_file():
+    log_path = f"{context.scenario.host_log_folder}/outputs/session_log.log"
```

Comment:
I have not been able to understand 100% of your changes, but I have been able to understand the intent of the PR.
I did a little test with the session_log file. You are generating a single file, but if we put a random file name, we can see that several files are generated (one per worker I guess). Now I wonder, this PR would eliminate 100% the collisions when overwriting the dockerfile or just reduce the probabilities?

` log_path = f"{context.scenario.host_log_folder}/outputs/session_log_{random.randint(0, 888889)}.log"`

Reply:
I have now changed this

Reply:
Now, we are generating one log per xdist worker, but it is only being written in one.
I can't see where the problem is in the code.

Reply:
Yes, that log file is only ever used during the building of the Docker image, and should only be written to by the worker actually building the docker image. Do you want me to change the name of the log files, or do you want there to be only one log file that the worker that does the building creates instead?

Reply:
I think that if we only write to one file, there should be only one.
Thanks @bantonsson ! I think this PR is a good improvement!

Reply:
Thanks, I'll fix that then.